### PR TITLE
Various optimizations

### PR DESCRIPTION
**Pull request by gomezzz**
# Description

Summary of changes

* Vectorized and optimized a bunch of functions in vegas

## How Has This Been Tested?

- [ x ] Ran vegas tests

### CODE REVIEW

File: torchquad/integration/vegas_map.py [hunk 1]
```diff
@@ -180,35 +180,30 @@ def update_map(
         """Update the adaptive map, Section II C."""
         self._smooth_map()
 
-        # Initialize new locations
```
Comment (GabrieleMeoni) [inline torchquad/integration/vegas_map.py:183]: Does it affect gradients?

Gradient tests fails for me for VEGAS():

grad=tensor([[-1.7841,  2.2152]])

gradient_tolerance = 5e-2

**Review by GabrieleMeoni — CHANGES_REQUESTED** [review]

File: torchquad/integration/vegas_map.py [hunk 2]
```diff
@@ -115,7 +115,7 @@ def accumulate_weight(self, y, jf_vec2):
             ID_i = torch.floor(ID[:, i]).long()
             unique_vals, unique_counts = torch.unique(ID_i, return_counts=True)
             weights_vals = jf_vec2
-            for val, count in zip(unique_vals, unique_counts):
+            for val, _ in zip(unique_vals, unique_counts):
```
[the remark this answers is not in the document]
Comment (gomezzz) [reply torchquad/integration/vegas_map.py]: Fixed

File: torchquad/integration/vegas_map.py (hunk 1 above)
Comment (gomezzz) [reply torchquad/integration/vegas_map.py:183]: The reason is likely that some of the parameters in VEGAS changed like N_strat. So for this particular test errors are a bit higher now. I increased N that reduces the error below the defined bounds, so I think it should be fine? (converges properly it seems)

Comment (gomezzz) [reply torchquad/integration/vegas_map.py:183]: For additional validation I am rerunning python vegas vs TQ vegas comparisons. Previous version 
image

Comment (gomezzz) [reply torchquad/integration/vegas_map.py:183]: New version 
image

Comment (gomezzz) [reply torchquad/integration/vegas_map.py:183]: So I think we are fine?

Comment (GabrieleMeoni) [reply torchquad/integration/vegas_map.py:183]: Yes, I don't have errors anymore.

**Review by GabrieleMeoni — APPROVED** [review]

**Pull request merged**